### PR TITLE
RELATED: CQ-670 - docs improvements

### PR DESCRIPTION
**Pull request by lupko**
- Added docs introducing and giving example how to use the TaskExecutor
- Discovered that the infra was missing some common convenience
- Added utilities / convenience methods to create TaskResult from Arrow Table or RecordBatchReader
- Added utility to create FlightDataStream for a task result; this can be used to conveniently implement do_get() when tasks are involved
- Removed some dead todos + clarified contract in FlexFun

### CODE REVIEW

File: gooddata-flexfun/gooddata_flexfun/flexfun/flex_fun.py [hunk 1]
```diff
@@ -16,16 +16,25 @@ class FlexFun(abc.ABC):
...
+    - the function receives `parameters` (arguments) that describe the context
+      in which it was invoked. The function _may_ use this information as input
+      to algorithms that it uses to generate the result data
 
-      NOTE: this is only about trimming columns - the result cardinality is
-      the same just that some columns are not returned.
+    - the caller may indicate to the function that out of all columns in the
+      schema, it is only interested in a subset of `columns`
+
+      NOTE: the list of `columns` is provided as hint column-trimming purposes.
```
Comment (no23reason) [inline gooddata-flexfun/gooddata_flexfun/flexfun/flex_fun.py:26]: Missing "for"

File: gooddata-flight-server/README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from typing import Anyfrom gooddata_flight_server import ServerContextfrom gooddata_flight_server import FlightDataTaskResultfrom gooddata_flight_server import TaskResultfrom typing import Union
```
Comment (no23reason) [inline gooddata-flight-server/README.md:1]: Why is this in the Readme?

Comment (lupko) [reply gooddata-flight-server/README.md:1]: beats me.. must have accidentally pasted or something..

File: gooddata-flexfun/gooddata_flexfun/flexfun/flex_fun.py (hunk 1 above)
Comment (lupko) [reply gooddata-flexfun/gooddata_flexfun/flexfun/flex_fun.py]: fixed

File: gooddata-flight-server/README.md (hunk 2 above)
Comment (lupko) [reply gooddata-flight-server/README.md]: fixed

**Review by no23reason — APPROVED** [review]

**Pull request merged**